src/libstd/sys/unix/ext.rs

```diff
@@ -130,19 +136,28 @@ impl OsStringExt for OsString {
     }
 }
 
-// Unix-specific extensions to `OsStr`.
+/// Unix-specific extensions to `OsStr`.
 pub trait OsStrExt {
-    fn from_byte_slice(slice: &[u8]) -> &OsStr;
-    fn as_byte_slice(&self) -> &[u8];
+    fn from_bytes(slice: &[u8]) -> &OsStr;
+
+    /// Get the underlying byte view of the `OsStr` slice.
+    fn as_bytes(&self) -> &[u8];
+
+    /// Convert the `OsStr` slice into a `CString`.
+    fn to_cstring(&self) -> CString;
```

Comment:
Incidentally, this is one reason that generic conversion traits are helpful (you don't have this kind of overlap worry).

But you're also raising an important general point, which is that we can't provide opt-in platform-specific impls for well-known traits.

Personally, I believe that this convenience is very important, because when you're working with `OsStr` on a given platform you need it to be very easy to convert that to something you can feed the OS. (I've already heard complaints about its lack). I'd somewhat prefer to land this now, but revisit before we stabilize.